### PR TITLE
Run detekt, klint, compare_locales against the expected commit

### PR DESCRIPTION
**Pull request by JohanLorenzo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] ~~**Changelog**: This PR includes a changelog entry or does not need one~~ Not applicable, it doesn't change the final product
- [ ] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~ Not applicable

While working on https://github.com/mozilla-mobile/fenix/pull/397, I realized detekt, klint, compare_locales are running against the revision baked in the docker image :scream:

### CODE REVIEW

**Comment by pocmo** [comment]
@JohanLorenzo 😱 Is that a recent regression? That used to work :)

File: automation/taskcluster/lib/tasks.py [hunk 1]
```diff
@@ -160,6 +149,16 @@ def _craft_build_ish_task(
         routes = [] if routes is None else routes
         features = {} if features is None else features
 
+        checkout_command = (
+            "export TERM=dumb && git fetch {} {} --tags && "
```
Comment (pocmo) [inline automation/taskcluster/lib/tasks.py]: FYI: "export TERM=dumb" has been set in our Docker images for a while already :)

**Comment by JohanLorenzo** [comment]
I introduced this 10 days ago in #1842 :disappointed:. I confirm that used to work thanks to https://github.com/mozilla-mobile/android-components/pull/1842/files#diff-168d551d2c0830884a5caeb50ae576e9L67

File: automation/taskcluster/lib/tasks.py (hunk 1 above)
Comment (JohanLorenzo) [reply automation/taskcluster/lib/tasks.py]: Oh right! Fixed

**Pull request merged**